stonesoup/models/measurement/nonlinear.py

```diff
@@ -965,3 +967,192 @@ def rvs(self, num_samples=1, **kwargs) -> Union[StateVector, StateVectors]:
+
+    .. math::
+
+      \vec{v}_t \sim \mathcal{N}(0,R)
+
+    .. math::
+
+      R = \begin{bmatrix}
+            \sigma_{\theta}^2 & 0 & 0 & 0\\
+            0 & \sigma_{\phi}^2 & 0 & 0\\
+            0 & 0 & \sigma_{r}^2 & 0\\
+            0 & 0 & 0 & \sigma_{\dot{r}}^2
+            \end{bmatrix}
+
+    The range and range rate are binned using
+
+    .. math::
+
+        x = floor(x/\Delta x)*\Delta x
+
+    The :py:attr:`mapping` property of the model is a 3 element vector, \
+    whose first (i.e. :py:attr:`mapping[0]`), second (i.e. \
+    :py:attr:`mapping[1]`) and third (i.e. :py:attr:`mapping[2]`) elements \
+    contain the state index of the :math:`x`, :math:`y` and :math:`z`  \
+    coordinates, respectively.
+
+    The :py:attr:`velocity_mapping` property of the model is a 3 element vector, \
+    whose first (i.e. :py:attr:`velocity_mapping[0]`), second (i.e. \
+    :py:attr:`velocity_mapping[1]`) and third (i.e. :py:attr:`velocity_mapping[2]`) elements \
+    contain the state index of the :math:`\dot{x}`, :math:`\dot{y}` and :math:`\dot{z}`  \
+    coordinates, respectively.
+
+    Note
+    ----
+    This class implementation assuming at 3D cartesian space, it therefore \
+    expects a 6D state space.
+    """
+
+    range_res = Property(float, doc="Size of the range bins")
+    range_rate_res = Property(float, doc="Size of the velocity bins")
+
+    @property
+    def ndim_meas(self):
+        return 4
+
+    def function(self, state, noise=False, **kwargs):
+        r"""Model function :math:`h(\vec{x}_t,\vec{v}_t)`
+
+        Parameters
+        ----------
+        state: :class:`~.StateVector`
+            An input state vector for the target
+
+        noise: :class:`numpy.ndarray` or bool
+            An externally generated random process noise sample (the default is
+            `False`, in which case no noise will be added and no binning takes place
+            if 'True', the output of :meth:`~.Model.rvs` is added and the
+            range and range rate are binned)
+
+        Returns
+        -------
+        :class:`numpy.ndarray` of shape (:py:attr:`~ndim_state`, 1)
+            The model function evaluated given the provided time interval.
+        """
+
+        out = super().function(state, noise, **kwargs)
+
+        if isinstance(noise, bool) or noise is None:
+            if noise:
+                out[0] = np.floor(out[0] / self.range_res) * self.range_res
+                out[1] = np.floor(
+                    out[1] / self.range_rate_res) * self.range_rate_res
+
+        return out
+
+    def _gaussian_integral(self, a, b, mean, cov):
+        # the cumlative probability ranging from a to b for a normal distribution
+        # \frac{1}{\sigma\sqrt{2\pi}}\int_{a}^{b}e^{-\frac{1}{2}\left(\frac{x-\mu}{\sigma}\right)^{2}}
+        return (multivariate_normal.cdf(a, mean=mean,
+                                        cov=cov) - multivariate_normal.cdf(
+            b, mean=mean, cov=cov))
```

Comment:
The indentation here can probably be improved

```suggestion
        return (multivariate_normal.cdf(a, mean=mean, cov=cov) 
               - multivariate_normal.cdf(b, mean=mean, cov=cov))
```